plugins/SitesManager/angularjs/sites-manager/sites-manager-type-model.js

```diff
@@ -14,11 +14,24 @@
             typesById: {},
             fetchTypeById: fetchTypeById,
             fetchAvailableTypes: fetchAvailableTypes,
-            hasMultipleTypes: hasMultipleTypes
+            hasMultipleTypes: hasMultipleTypes,
+            removeEditSiteIdParameterFromHash: removeEditSiteIdParameterFromHash,
+            getEditSiteIdParameter: getEditSiteIdParameter
         };
 
         return model;
 
+        function getEditSiteIdParameter() {
+            var search = String(window.location.hash).substr('#/'.length);
+            var searchParams = piwik.helper.getArrayFromQueryString(search);
+            if (searchParams.editsiteid && $.isNumeric(searchParams.editsiteid)) {
+                return searchParams.editsiteid;
+            }
+        }
+        function removeEditSiteIdParameterFromHash() {
+            window.location.hash = window.location.hash.replace(/editsiteid=\d+/g, '');
```

Comment:
This could end up w/ an extra `&`, correct? Eg, `matomo.com/?date=abc&editsiteid=123` => `matomo.com/?date=abc&`. Added a suggestion to fix.
```suggestion
            window.location.hash = window.location.hash.replace(/[&?]editsiteid=\d+/g, '');
```

Reply:
The hash I'm using currently would be `...#/editsiteid=5` so it shouldn't be a problem?

Reply:
Oh I see, it's the admin page so we don't have to worry about the hash having a query string, :+1: